src/main/java/dev/mccue/resolve/core/CacheHandler.java

```diff
@@ -0,0 +1,84 @@
+package dev.mccue.resolve.core;
+
+import java.io.FileOutputStream;
+import java.net.URL;
+import java.nio.channels.Channels;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Locale;
+
+public final class CacheHandler {
+    public enum OSLabel {
+        Linux, MacOS, Windows
+    }
+    private static OSLabel currentOS;
+    private static String CACHE_PATH;
+
+    public CacheHandler() {
+        /* This only handles the default cache path and will need to
+        *  be modified when new features are implemented (e.g. CLI --cache option)
+        */
+        checkOS();
+        switch (currentOS){
+            case Linux:
+                CACHE_PATH = System.getProperty("user.home") + "/.cache/jresolve";
+                break;
+            case MacOS:
+                CACHE_PATH = System.getProperty("user.home") + "/Library/Caches/JResolve";
+                break;
+            case Windows:
+                CACHE_PATH = System.getenv("APPDATA") + "\\JResolve\\Cache";
+                break;
+            case null:
+                throw new IllegalArgumentException("Operating System could not be determined.");
+            case default:
+                break;
+        }
+    }
+
+    public void cacheAdd(String url) throws Exception {
+        String[] parsedURL = URLParser(url);
+        StringBuilder path;
+        if (currentOS != OSLabel.Windows) {
+            path = new StringBuilder(CACHE_PATH + "/" + parsedURL[0] + "/");
+            for(int i = 1; i < parsedURL.length-1; i++) {
+                path.append(parsedURL[i]).append("/");
+            }
+        } else {
+            path = new StringBuilder(CACHE_PATH + "\\" + parsedURL[0] + "\\");
+            for(int i = 1; i < parsedURL.length-1; i++) {
+                path.append(parsedURL[i]).append("\\");
+            }
+        }
```

Comment:
This exact logic is what the `Path` class is for. It handles platform specific separators in paths.